### DISCUSSION TITLE
Features request … for the future

### DISCUSSION THREAD

**Discussion by Jean-Gabriel1982** [GrandOrgue/grandorgue#785] [General]
Dear all,
A organ seller working with Hauptwerk is very interested in using GrandOrgue for his customers. He asked me to post this feature request and kindly ask if those points are realisable or out of mind ;)

I think some of those features would be very nice for the Future of GrandOrgue. 

Here is the message :

- Harmonization: with HW we basically use three functions to correct the acoustic aberrations note by note, due to the system and especially to the location: the volume of each note (already existing on GO), the "Brightness" function which seems to play on the rate of a complex of harmonics from the fundamental of the processed note, and finally the "high frequency" function which is nothing other than an equalizer which plays on very high frequencies. We should be able to copy these functions on GO and ideally create an equalizer where we can define the bandwidth (s) that we want to process (less fundamental function)

-A system that allows each stop to be distributed over a number of channels greater than 2. In practice, 20 channels maximum would already be enormous, I do not think that we will ever have the opportunity to sell an installation with 20 speakers ... perhaps to go further than our current practice of HW, a system that can allow a subwoofer to be used on several channels at the same time. So as not to need a box per pair of speakers, but for example a box for 4 or 6 speakers (less fundamental function at first)

-The possibility of having a system which accepts a certain number of HW sound banks (mainly sonus)
(My comment : no possible as HW sound banks are not public domaine. You have not to answer this point) 😀 

-A parameter setting mode with specific access for the technician (by a code, a keyboard shortcut ...), and a user mode with a simple display of the essential functions (minimalist display of the stops, access combiner, transpositor, choice of organs, temperaments ...)

-A global equalizer would not be bad too, as well as a qualitative reverb (but secondly, you can do without)

**Comment by larspalo** [comment]
We don't (yet) have any equalizer feature at all in GO. It's not impossible that at some point in the future we'll have that kind of voicing capability in GO, but at the moment - no, you'll have to resort to using another software solution in between GO and sound output. The only effort to include any kind of equalizer feature was when last year I experimented with adding a shelf filter as swell box model  #717 that is in effect a much more sophisticated implementation than the current amplitude only.

It's already possible to use the audio group feature to distribute the channels in any way imaginable. It's mainly limited by the possibility of the audio device (it's number of output channels), as using more than one is not recommended. Besides that, GO doesn't impose any limits to the number of speakers.

It's certainly possible to create a hardware solution package to sell that run GO within. In such a product it's of course possible to (for the seller) buy any number of samplesets (with proper license from copyright holders) that will be available with the end product for the user to use. However, there's no specific solution as to how to create such a product - that will be up to the vendor wanting to sell it to manage.

A proper setup of midi messages will be able to control a lot of the built in features of GO. Display can be also controlled by messages, or by specific panels - it all depends on what type is used and it's limitations.

Artificial reverb can be added in a multitude of ways, from within GO by impulse responses and by external tools in whatever way you want (you just route the audio signal properly).

**Comment by oleg68** [comment]
> It's already possible to use the audio group feature to distribute the channels in any way imaginable. It's mainly limited by the possibility of the audio device (it's number of output channels)

The best way to do it - to one virtual device `Jack Native Output` with as many channels as need and distribute its ports across many physical audio devices and even programs sound processors.

> using more than one is not recommended.

I think this recommendation has been caused by the unstable mutex implementation in GO. I hope after reimplementing it in #702 we will be able to withdrow this recommendation.
